sources/packages/api/src/services/designation-agreement/designation-agreement-locations.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from "@nestjs/common";
+import { RecordDataModelService } from "../../database/data.model.service";
+import { Connection, SelectQueryBuilder } from "typeorm";
+import { DesignationAgreementLocation } from "../../database/entities";
+
+/**
+ * Manages the operations needed for designation agreements location.
+ */
+@Injectable()
+export class DesignationAgreementLocationService extends RecordDataModelService<DesignationAgreementLocation> {
+  constructor(private readonly connection: Connection) {
+    super(connection.getRepository(DesignationAgreementLocation));
+  }
+
+  /**
+   * Creates a 'select' query that could be used in an 'exists' or
+   * 'not exists' where clause to define if the location is designated
+   * or not.
+   * ! This query will assume that a join to 'institution_location.id' is present
+   * ! in the master query.
+   * @returns 'select' query that could be used in an 'exists' or
+   * 'not exists'.
+   */
+  getExistsDesignatedLocation(): SelectQueryBuilder<DesignationAgreementLocation> {
+    return this.repo
+      .createQueryBuilder("designationAgreementLocation")
+      .select("1")
+      .innerJoin(
+        "designationAgreementLocation.designationAgreement",
+        "designationAgreement",
+      )
+      .where("designationAgreement.designationStatus = 'Approved'")
```

Comment:
for designationStatus we can use DesignationAgreementStatus enum here.

Reply:
it was changed 
![image](https://user-images.githubusercontent.com/77353155/158903499-48613815-8321-4b78-abdb-0d5c340a0838.png)
@andrewsignori-aot

Reply:
@ann-aot my comment was about the external parameter we can still use the enum as below 😉 
```
.where(`designationAgreement.designationStatus = '${DesignationAgreementStatus.Approved}'`)
```